framework/src/utils/RankTwoTensor.C

```diff
@@ -891,11 +891,11 @@ RankTwoTensorTempl<T>::symmetricEigenvaluesEigenvectors(std::vector<T> & eigvals
 
 template <typename T>
 void
-RankTwoTensorTempl<T>::adSymmetricEigenvaluesEigenvectors(RankTwoTensorTempl<T> & Q,
-                                                          RankTwoTensorTempl<T> & D) const
+RankTwoTensorTempl<T>::symmetricEigenvaluesEigenvectors(RankTwoTensorTempl<T> & Q,
```

Comment:
No, this is not a template specialization.

Reply:
And why does this have a different signature than the other method?

Reply:
The decomposition is written as A = Q D Q^T, and some of the dual numbers reside in off-diagonal entries of the diagonal matrix D. I know ideally there shouldn't be any off-diagonal dual numbers, but all practical algorithms I have tried give me non-zero off-diagonal dual numbers.

Therefore, I am requesting and returning not just std::vector<DualReal> & eigvals, but the entire eigenvalue matrix RankTwoTensorTempl<T> & D. That's why I am writing it as a function overloading rather than a template specialization right now.